src/pages/main/tools/Editor/Editor.tsx

```diff
@@ -0,0 +1,68 @@
+import Draggable, { DraggableData, DraggableEvent } from 'react-draggable';
+import { LatexTextArea, PdfViewer } from '@components';
+
+import { RxDotsVertical } from 'react-icons/rx';
+import styles from './Editor.module.less';
+import { useResizeDetector } from 'react-resize-detector';
+import { useState } from 'react';
+
+const Editor = () => {
+  const DELIMITER_WIDTH = 5;
+  const MIN_PERCENTAGE_WIDTH = 0.3;
+
+  // Handle initial delimiter position
+  const { width, height, ref: rootRef } = useResizeDetector();
+  const [defaultDelimiterX, setDefaultDelimiterX] = useState(0.5);
+
+  const onStop = (_: DraggableEvent, data: DraggableData) => {
+    setDefaultDelimiterX(data.x / width);
+  };
+
+  interface DelimiterProps {
+    testId: string
+  }
+
+  const Delimiter = (props: DelimiterProps) => {
+    return (
+      <Draggable 
+        axis='x'
+        offsetParent={rootRef.current}
+        defaultPosition={{ x: defaultDelimiterX * width - DELIMITER_WIDTH/2, y: 0 }}
```

Comment:
please yeet all these calculations to some states and update them using `useEffect`, e.g.

```
const [defaultDelimiterPosition, setDefaultDelimiterPosition] = useState<number>(0);
/* ... other similar states */

useEffect(() => {
 /* update all states after width change */
}, [width]);
```